lib/queries/batchUpsert.js

```diff
@@ -1,32 +1,43 @@
 import valueSubQuery from './helpers/valueSubQuery';
 import batchParameter from './helpers/batchParameter';
+import sanitizeIdentifier from './helpers/sanitizeIdentifier';
+import whereQuery from './helpers/whereQuery';
 
 export default ({
     table,
     primaryKey,
     writableCols,
     returnCols = ['*'],
+    permanentFilters = {},
 }) => (rows) => {
-    const columns = primaryKey.concat(writableCols.filter((f) => primaryKey.indexOf(f) === -1));
+    const columns = primaryKey.concat(writableCols.filter(f => primaryKey.indexOf(f) === -1));
     const getValueSubQuery = valueSubQuery(columns);
     const getParameter = batchParameter(columns);
 
     const returning = returnCols.join(', ');
 
-    const setQuery = writableCols.map((col) => `${col} = excluded.${col}`);
+    const setQuery = writableCols.map(col => `${col} = excluded.${col}`);
 
     const values = rows
         .map((_, index) => getValueSubQuery(index + 1))
         .reduce((result, value) => result.concat(`(${value})`), []);
 
     const parameters = getParameter(rows);
 
+    const permanentFiltersKeys = Object.keys(permanentFilters);
+    let where = '';
+    let permanentParameters = {};
+    if (permanentFiltersKeys.length) {
+        permanentParameters = sanitizeIdentifier(permanentFiltersKeys, permanentFilters);
+        where = whereQuery(permanentParameters, permanentFiltersKeys);
+    }
+
     const sql = (
 `INSERT INTO ${table}
 (${columns.join(', ')})
 VALUES ${values.join(', ')}
 ON CONFLICT (${primaryKey.join(', ')})
-DO UPDATE SET ${setQuery.join(', ')}
+DO UPDATE SET ${setQuery.join(', ')} ${where}
```

Comment:
```js
where ? ` ${where}` : ''
```

Reply:
ternary operator is applied  [here](https://github.com/marmelab/coPostgresQuery/pull/48/commits/10f21729a60ca63617cc0fc21390e8c5cc7c91a2#diff-d514f10459a4924ed38247e4be45958bR27) (line 27)